src/workerd/api/memory-cache.h

```diff
@@ -434,7 +434,8 @@ class MemoryCache: public jsg::Object {
 // uses a simple in-memory map to store the SharedMemoryCache instances.
 class MemoryCacheProvider {
 public:
-  virtual SharedMemoryCache& getInstance(kj::StringPtr cacheId) const = 0;
+  virtual ~MemoryCacheProvider() noexcept(false) = default;
+  virtual SharedMemoryCache& getInstance(kj::StringPtr cacheId, uint32_t ownerId) const = 0;
```

Comment:
I believe that the way this already works internally is that the cache IDs are globally unique. There's no need to add `ownerId`, and in fact this might break cases where a cache is intentionally shared between owners (I don't know if any such cases exist today).

Reply:
The keys are globally unique, yes, but it doesn't appear to be any form of validation. That is, if I'm lucky enough to guess an existing key in the binding configuration then I can get access to that existing cache. But this is something that needs to be evaluated before this lands... it's likely there's a very different way we should handle this.

Reply:
You mean the control plane API lets you specify any ID you want today? That sounds like a control plane issue which we would indeed need to fix before this is open to the public.

Reply:
No idea what the control plane allows and I want to make sure that we are resilient that. If we are reasonably certain that the control plane ensures that these values are not arbitrarily passed through then we might be ok, but I'd want some verification of that to be on the safe side.

Reply:
ok, in offline communication we decide that it is ultimately the role of the control plane to ensure the uniqueness guarantee of a cache id. That's something we will follow up with separately. For now, I'll be changing this back to remove the additional `ownerId` arg.